File: frontend/src/services/SubCategoryService.js
import API from "./AuthService";



/** Add SubCategory */
export const addSubCategory = (token, data) => {
  return API.post(API.BASE_URL, data, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

/** Get All SubCategories */
export const getAllSubCategories = (token) => {
  return API.get(API.BASE_URL, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

/** Get SubCategories by Category ID */
export const getByCategoryId = (token, categoryId) => {
  return API.get(`${API.BASE_URL}/category/${categoryId}`, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

/** Update SubCategory */
export const updateSubCategory = (token, id, data) => {
  return API.put(`${API.BASE_URL}/${id}`, data, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

/** Delete SubCategory */
export const deleteSubCategory = (token, id) => {
  return API.delete(`${API.BASE_URL}/${id}`, {
    headers: { Authorization: `Bearer ${token}` },
  });
};

File: src/main/java/ahc/dms/dao/dms/services/SubCategoryService.java
package ahc.dms.dao.dms.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import ahc.dms.dao.dms.entities.SubCategory;
import ahc.dms.dao.dms.repositories.SubCategoryRepository;

@Service
public class SubCategoryService {

    @Autowired
    private SubCategoryRepository subCategoryRepository;

    // Create / Add
    public SubCategory addSubCategory(SubCategory subCategory) {
        return subCategoryRepository.save(subCategory);
    }

    // Get all
    public List<SubCategory> getAllSubCategories() {
        return subCategoryRepository.findAll();
    }

    // Get by Category ID
    public List<SubCategory> getSubCategoriesByCategory(Long categoryId) {
        return subCategoryRepository.findByCategoryId(categoryId);
    }

    // Update
    public SubCategory updateSubCategory(Long id, SubCategory updated) {
        SubCategory sc = subCategoryRepository.findById(id).orElseThrow();
        sc.setName(updated.getName());
      
        sc.setCategory(updated.getCategory());
        return subCategoryRepository.save(sc);
    }

    // Delete
    public void deleteSubCategory(Long id) {
        subCategoryRepository.deleteById(id);
    }
}
